Sources/XiEditor/LineCache.swift

```diff
@@ -60,6 +54,41 @@ struct Line<T> {
     }
 }
 
+enum AnnotationType: String {
+    case Selection = "selection"
+    case Highlight = "highlight"
+}
+
+extension AnnotationType {
+    static var all: [AnnotationType] {
+        return [AnnotationType.Selection, AnnotationType.Highlight]
+    }
+}
+
+/// Represents an annotation (eg. selection, find highlight)
+struct Annotation {
+    var startLine: Int
+    var startColumn: Int
+    var endLine: Int
+    var endColumn: Int
+    var metadata: [String: AnyObject]?
+
+    init(fromJson json: [String: AnyObject]) {
```

Comment:
If a specific requirements need to be met in order to create `Annotation` we could make use of the optional initializer here 😄

```swift
    init?(fromJson json: [String: AnyObject]) {
         guard let position = json["pos"] as? [Int], position.count == 4 else { return nil }

         startLine = position[0]
         startColumn = position[1]
         endLine = position[2]
         endColumn = position[3]
         metadata = json["metadata"] as? [String: AnyObject]
     }
```